fdbserver/storageserver.actor.cpp

```diff
@@ -2069,8 +2090,14 @@ ACTOR Future<Void> getValueQ(StorageServer* data, GetValueRequest req) {
 			                      req.options.get().debugID.get().first(),
 			                      "getValueQ.AfterVersion"); //.detail("TaskID", g_network->getCurrentTask());
 
-		data->checkTenantEntry(version, req.tenantInfo);
 		if (req.tenantInfo.hasTenant()) {
+			bool lockAware = req.options.present() && req.options.get().lockAware;
+			auto tenant = data->getTenantEntry(req.version, req.tenantInfo);
+			if (!tenant.present()) {
```

Comment:
I think this isn't possible; if a tenant is specified in tenantInfo then it will return the entry or an error.

Reply:
Ok, I will remove this if-branch